src/memkind_memtier.c

```diff
@@ -1189,14 +1192,63 @@ MEMKIND_EXPORT void *memtier_malloc(struct memtier_memory *memory, size_t size)
     void *ptr;
     uint64_t data;
 
-    ptr = memtier_kind_malloc(memory->get_kind(memory, size, &data), size);
+    dont_mmap = true;
+    memkind_t kind = memory->get_kind(memory, size, &data);
+    ptr = memtier_kind_malloc(kind, size);
     memory->post_alloc(data, ptr, size);
     memory->update_cfg(memory);
     print_memory_statistics(memory);
 
+    dont_mmap = false;
     return ptr;
 }
 
+void *sys_mmap(void *addr, size_t length, int prot, int flags, int fd, off_t off)
+{
+    long ret = syscall(SYS_mmap, addr, length, prot, flags, fd, off);
+    if (ret == -EPERM && !off && (flags&MAP_ANON) && !(flags&MAP_FIXED))
+        ret = -ENOMEM;
+    if (ret > -4096 && ret < 0) {
+        errno = -ret;
+        return MAP_FAILED;
+    }
+
+    return (void*)ret;
+}
+
+int munmap(void *addr, size_t length)
+{
+    long ret = syscall(SYS_munmap, addr, length);
+    if (!ret)
+        return 0;
+    errno = -ret;
+    return -1;
+}
+
+#include <pthread.h>
+pthread_mutex_t mutex;
```

Comment:
Not static... Is it ever used as "extern"? if yes, this is probably not the best way to use a mutex, I've seen horrible things happen because of such approach - ugly code, synchronization errors and deadlocks, among others... It would be best to:

1. Describe what this mutex protects exactly, preferrably by giving it self-explanatory name
2. Keep the data it protects (if it's about data access synchronisation) as close to it as possible - preferrably in the same structure